Fix Keyboard.Custom.Clone to make sure Colors for Keys are cloned with the Key Flag

WolfspiritM opened the pull request:
This PR fixes Custom.Clone for Keyboard.
Right now the following won't work:
``` c#
	var original = new Corale.Colore.Razer.Keyboard.Effects.Custom(Color.Red)
	{
		[Corale.Colore.Razer.Keyboard.Key.A] = Color.Green,
		[0, 0] = Color.Orange
	};
	var c2 = original.Clone();
	Chroma.Instance.Keyboard.SetCustom(c2);
```
as it will set everything to red and Key.A will also be red.
Reason is that the clone function sets:
``` c#
_keys[index] = other[(Key)index];
```
While Custom[index] will return the Color without the Key-Flag.
The solution is to always add the Key-Flag to the Value if there is a color set ( Value > 0 )

I also made the Test more "usefull" by checking the private Fields via Reflection if they are 100% equal.

njbmartin commented:
Nice work!

Sharparam commented on Corale.Colore/Razer/Keyboard/Effects/Custom.cs:
```diff
@@ -86,7 +86,11 @@ public Custom(Custom other)
+                var otherKeyColor = other[(Key)index];
+                if (otherKeyColor.Value > 0)
+                {
+                    _keys[index] = Constants.KeyFlag | otherKeyColor;
```
This will force the key enabled, even if previously disabled.

E.g.: `0x00FFFFFF` would be copied as `0x01FFFFFF`.

Probably better to use raw value here as well.

WolfspiritM commented:
I agree!
I changed that and also added a testcase for that (Escape Key gets enabled and disabled via accessor). That test case failed with my previous implementation and works with the direct assignment via private members.

Sharparam commented:
You'll have to rebase your branch since there was a PR merged earlier. (There should be an "Update branch" button at the bottom here.)
